src/attachment.rs

```diff
@@ -97,16 +107,23 @@ impl Attachment {
             .unwrap_or("file")
             .to_owned();
 
-        let file = File::open(path).await?;
+        let file = File::open(path.clone()).await?;
         let content_length = file.metadata().await?.len();
         let stream = Body::wrap_stream(FramedRead::new(file, BytesCodec::new()));
```

Comment:
An alternative implementation could use [`imagesize::reader_size`](https://docs.rs/imagesize/latest/imagesize/fn.reader_size.html) with `&mut file` here, before moving the file into this `Body`.